identity/src/main/java/com/stripe/android/identity/navigation/SelfieFragment.kt

```diff
@@ -144,129 +126,73 @@ internal class SelfieFragment(
     }
 
-    override fun updateUI(identityScanState: IdentityScanState) {
-        when (identityScanState) {
-            is IdentityScanState.Initial -> {
-                // no-op
-            }
-            is IdentityScanState.Found -> {
-                maybeFlash()
-                binding.message.text = requireContext().getText(R.string.capturing)
-            }
-            is IdentityScanState.Unsatisfied -> {} // no-op
-            is IdentityScanState.Satisfied -> {
-                binding.message.text = requireContext().getText(R.string.selfie_capture_complete)
-            }
-            is IdentityScanState.Finished -> {
-                binding.message.text = requireContext().getText(R.string.selfie_capture_complete)
-                toggleResultViewWithResult(
-                    (identityScanState.transitioner as FaceDetectorTransitioner)
-                        .filteredFrames.map { it.first.cameraPreviewImage.image.mirrorHorizontally() }
-                )
-            }
-            is IdentityScanState.TimeOut -> {
-                // no-op, transitions to CouldNotCaptureFragment
-            }
-        }
-    }
-
-    override fun resetUI() {
-        scanningView.visibility = View.VISIBLE
-        padding.visibility = View.VISIBLE
-        resultView.visibility = View.GONE
-        continueButton.isEnabled = false
-        messageView.text = requireContext().getText(R.string.position_selfie)
-    }
-
     /**
      * Collect the [IdentityViewModel.selfieUploadState] and update UI accordingly.
      *
      * Try to [postVerificationPageDataAndMaybeSubmit] when all images are uploaded and navigates
      * to error when error occurs.
      */
-    private fun collectUploadedStateAndUploadForCollectedSelfies() =
+    @VisibleForTesting
+    internal fun collectUploadedStateAndUploadForCollectedSelfies(allowImageCollection: Boolean) =
         lifecycleScope.launch {
-            viewLifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
-                identityViewModel.selfieUploadState.collectLatest {
-                    when {
-                        it.hasError() -> {
-                            "Fail to upload files: ${it.getError()}".let { msg ->
-                                Log.e(TAG, msg)
-                                navigateToDefaultErrorFragment(msg)
-                            }
-                        }
-                        it.isAnyLoading() -> {
-                            continueButton.toggleToLoading()
+            identityViewModel.selfieUploadState.collectLatest {
+                when {
+                    it.isIdle() -> {} // no-op
+                    it.isAnyLoading() -> {} // no-op
+                    it.hasError() -> {
+                        "Fail to upload files: ${it.getError()}".let { msg ->
+                            Log.e(TAG, msg)
+                            navigateToDefaultErrorFragment(msg)
                         }
```

Comment:
we're intentionally using `collectLatest` here right? Doesn't look like it, but if we're expecting fast consequent emissions on selfieUploadState we might want to use `collect`.

Reply:
yes it's intentional, we'll ignore the previous state updates